### PR TITLE
Dev DB update workflow 

### PR DESCRIPTION
**Pull request by henry-li-06**
### Summary 

This PR adds a new GitHub Actions workflow that executes a script to update the dev Firebase instance. The script essentially overwrites the entire dev db with the data from the prod db. The script runs every month on the first day of the month at 3-4 AM (depending on daylight savings) or can be executed manually. 

### Test Plan 
Run the script locally

### CODE REVIEW

**Review by riyajaggi — COMMENTED** [review]
There are a couple of GitHub actions check warnings on the PR. It might be cleaner to resolve them before merging this PR.

image

**Comment by henry-li-06** [comment]
> There are a couple of GitHub actions check warnings on the PR. It might be cleaner to resolve them before merging this PR.
> 
> image

@riyajaggi fixed! thanks for catching this

File: backend/scripts/update-dev-db.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      credential: admin.credential.cert(configureAccount(prodServiceAccount, true)),
+      databaseURL: 'https://idol-b6c68.firebaseio.com',
+      storageBucket: 'gs://cornelldti-idol.appspot.com'
+    },
+    'prod'
+  );
+  const devApp = admin.initializeApp(
+    {
+      credential: admin.credential.cert(configureAccount(devServiceAccount, false)),
+      databaseURL: 'https://idol-b6c68.firebaseio.com',
```
Comment (JacksonStaniec) [inline backend/scripts/update-dev-db.ts:111]: Is this url supposed to be different from the one for the prod db? Looking at the firebase console, the dev db has a different url, but it seems to be a weird default one.

Comment (henry-li-06) [reply backend/scripts/update-dev-db.ts:111]: Tbh I'm not even 100% if it needs to be included. I think this db URL is referring to firebase's realtime database which is different than the firestore database. We don't even ever use this feature of firebase.

**Review by JacksonStaniec — APPROVED** [review]
No more outdated docs messing up testing!

**Pull request merged**